### PR TITLE
Add serialization tests for UTCTime

### PR DESCRIPTION
**Pull request by DavidEichmann**
I've split into 2 commits for clarity.

### CODE REVIEW

**Comment by DavidEichmann** [comment]
Does anyone know why we seem to have 2 `.cabal` files that declare the same testsuite?

**Review by DeepakKapiswe — APPROVED** [review]
LGTM

File: binary/test/Test/Cardano/Binary/Serialization.hs [hunk 1]
```diff
@@ -147,6 +152,26 @@ instance FromCBOR TestStruct where
...
+    Gen.integral (Range.constantFrom 0 0 ((86401e12) - 1))
+
+genDay :: Gen Time.Day
+genDay = Time.fromOrdinalDate <$> genYear <*> genDayOfYear
+
+genYear :: Gen Integer
+genYear = Gen.integral (Range.linear (-10000) 10000)
+
+genDayOfYear :: Gen Int
+genDayOfYear = Gen.int (Range.constantFrom 1 1 366)
```
Comment (newhoggy) [inline binary/test/Test/Cardano/Binary/Serialization.hs:174]: What happens in leap years?

**Review by newhoggy — APPROVED** [review]

**Review by newhoggy — APPROVED** [review]

File: binary/test/Test/Cardano/Binary/Serialization.hs (hunk 1 above)
Comment (DavidEichmann) [reply binary/test/Test/Cardano/Binary/Serialization.hs:174]: Yes, this is perhaps a bit subtle, but `fromOrdinalDate` should take care of clipping this in the case of non-leap years. I figured the random distribution having that small quirk was not worth fixing with extra complexity in the generator.

**Comment by DavidEichmann** [comment]
bors merge

**Pull request merged**